src/include/migraphx/match/gelu_tanh.hpp

```diff
@@ -36,31 +36,44 @@ template <class F>
 struct gelu_tanh_matcher
 {
     F f;
-    auto pow_fn() const { return f("pow")(used_once(), arg(1)(has_value(3.0f))); }
 
+    /// x ^ 3
+    auto pow_fn() const { return f("pow")(used_once(), arg(1)(has_value(3.0f, 10, 10))); }
+
+    /// tanh( sqrt(2/M_PI) * (x + 0.044715 * x ^ 3 )
     auto tanh_fn() const
     {
-        return f("tanh")(
-            used_once(),
-            arg(0)(f("mul")(either_arg(0, 1)(has_value(sqrt(M_2_PI), 1e-3),
-                                             f("add")(any_arg(0, 1)(f("mul")(either_arg(0, 1)(
-                                                 has_value(0.044715f), pow_fn()))))))));
+        auto mul_const_pow = f("mul")(either_arg(0, 1)(has_value(0.044715f, 10, 10), pow_fn()));
+        auto add_any_mul         = f("add")(any_arg(0, 1)(mul_const_pow));
+        auto either_SQRT2RPI_add = either_arg(0, 1)(has_value(sqrt(M_2_PI), 10, 10), add_any_mul);
+        return f("tanh")(used_once(), arg(0)(f("mul")(either_SQRT2RPI_add)));
+    }
+
+    /// x * (0.5? + 0.5 * tanh( sqrt(2/M_PI) * (x? + 0.044715 * x? ^ 3) ) )
+    /// <item>? question mark means it doesn't explicitly match that item (anything will work)
+    auto matcher_v0() const
```

Comment:
the V0 version is what used to be there. It doesn't match to anything that I know of since we were not using it before. The V1 version is what I found in distilgpt2

Reply:
Can we just combine them with:

```cpp
auto matcher() const
{
    return any_of(matcher_v0(), matcher_v1());
}
```

?